site/src/Librecores/ProjectRepoBundle/Doctrine/ProjectMetricsProvider.php

```diff
@@ -238,4 +267,290 @@ function (LanguageStat $a, LanguageStat $b) {
+        }
+
+        // +1 for a license file
+        // -3 since this is very important for open source projects
+        if ($project->getLicenseText()) {
+            $score += 1;
+        } else {
+            $score -= 3;
+        }
+
+        // TODO: This section needs improvement
+
+        // commit activity
+        $commitActivity = $this->getCommitActivityTrend($project);
+        $averageCommits = StatsUtil::normalize($this->getCommitCountAverage($project));
+
+        // +0.5 for stable project with mid term > 0.2
+        if ($averageCommits['mid'] > 0.2) {
+            $score += 0.5;
+        } else {
+            $score -= 0.5;
+        }
+
+        // +0.5 for constant commits
+        if ($averageCommits['end'] > 0.05) {
+            $score += 0.5;
+        } else {
+            $score -= 0.5;
+        }
+
+        // +0.5 if commit activity is not decreasing
+
+        if ($commitActivity['mid'] >= 0) {
+            $score += 0.25;
+        } else {
+            $score -= 0.25;
+        }
+
+        if ($commitActivity['end'] >= 0) {
+            $score += 0.25;
+        } else {
+            $score -= 0.25;
+        }
+
+        // +0.5 if project is still interesting
+
+        if ($this->getContributorActivityTrend($project) > 0.3) {
+            $score += 0.5;
+        }
+
+        return max(ceil($score * 5/17.5), 0);
+    }
+
+
+    /**
+     * Calculate trends in commit activity
+     *
+     * @param Project $project
+     * @return array
+     */
+    public function getCommitActivityTrend(Project $project)
+    {
+        $yearlyCommitCount = array_values($this->commitRepository->
+            getCommitHistogram($project->getSourceRepo(), Dates::INTERVAL_YEAR));
+
+        if (empty($yearlyCommitCount)) {
+
+            return [
+                'start' => -1,
+                'mid' => -1,
+                'end' => -1
+            ];
+        }
+
+        $length = count($yearlyCommitCount);
+
+        // It makes sense to divide a 6 year old project not a young project
+        if ($length >= 6) {
+            list($start, $mid, $end) = array_chunk($yearlyCommitCount, ceil(count($yearlyCommitCount) / 3));
+
+            return [
+                'start' => StatsUtil::averageRateOfChange($start),
+                'mid' => StatsUtil::averageRateOfChange($mid),
+                'end' => StatsUtil::averageRateOfChange($end)
+            ];
+        } else if ($length > 2) {
+            list($start, $mid) = array_chunk($yearlyCommitCount, ceil(count($yearlyCommitCount) / 2));
+
+            return [
+                'start' => StatsUtil::averageRateOfChange($start),
+                'mid' => StatsUtil::averageRateOfChange($mid),
+                'end' => -1
+            ];
+        }
+
+        return [
+            'start' => StatsUtil::averageRateOfChange($yearlyCommitCount),
+            'mid' => -1,
+            'end' => -1,
+        ];
+    }
+
+    /**
+     * Calculate trends in commit activity
```

Comment:
Please add a note about what this function does, and what its limitations are as discussed on Tuesday.